src/allmydata/storage_client.py

```diff
@@ -1271,6 +1271,11 @@ def stopService(self):
         if self._lc.running:
             self._lc.stop()
         self._failed_to_connect("shut down")
+
+        maybe_storage_server = self.get_storage_server()
+        if maybe_storage_server is not None:
+            result.addCallback(lambda _: maybe_storage_server._http_client.shutdown())
```

Comment:
I don't know whether `get_storage_server` could change its return value between when line 1275 runs and when the lambda in line 1277 runs.  And I'm confused by the implementation of `get_storage_server` itself now.  Thoughts?

A bit separately - since this code is `HTTPNativeStorageServer` itself and only works if `get_storage_server` is going to return an `_HTTPStorageServer`, maybe it should just look at `self._istorage_server` directly instead of going through `self.get_storage_server`?

Reply:
the `_istorage_server` is only ever set once, so it won't change once it's set the first time. At least, for now.